trio/testing/_check_streams.py

```diff
@@ -2,24 +2,33 @@
 from __future__ import annotations
 
 import random
+from collections.abc import Awaitable, Callable, Generator
 from contextlib import contextmanager
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Generic, TypeVar
 
-from .. import _core
-from .._abc import HalfCloseableStream, ReceiveStream, SendStream, Stream
+from .. import CancelScope, _core
+from .._abc import AsyncResource, HalfCloseableStream, ReceiveStream, SendStream, Stream
 from .._highlevel_generic import aclose_forcefully
 from ._checkpoints import assert_checkpoints
 
 if TYPE_CHECKING:
     from types import TracebackType
 
+    from typing_extensions import ParamSpec, TypeAlias
 
-class _ForceCloseBoth:
-    def __init__(self, both):
-        self._both = list(both)
+    ArgsT = ParamSpec("ArgsT")
 
-    async def __aenter__(self):
-        return self._both
+Res1 = TypeVar("Res1", bound=AsyncResource)
+Res2 = TypeVar("Res2", bound=AsyncResource)
+StreamMaker: TypeAlias = "Callable[[], Awaitable[tuple[Res1, Res2]]]"
```

Comment:
If this is a string only because of `tuple`, maybe use capital-T `Tuple`? (I've been so out of it I forgot what the convention here is...)

Reply:
It's actually because of Callable + Awaitable too.

Reply:
Oh, is this maybe cause docs?